lib/node_starter/queue_subscribe.rb

```diff
@@ -64,21 +54,24 @@ def run(delivery_info, payload)
         params['enqueue_data'],
         node_api_address(uuid))
 
-      begin
-        starter.start_node_process
-        @consumer.ack(delivery_info)
-        @shutdown_consumer.register_node(params['build_id'])
-      rescue => e
-        NodeStarter.logger.error "Node #{params['build_id']} spawn failed: #{e}"
-        @consumer.reject(delivery_info, true)
-        return
-      end
+      @reporting_publisher.receive(params['build_id'])
+      starter.start_node_process
+      @consumer.ack(delivery_info)
+      @shutdown_consumer.register_node(params['build_id'])
```

Comment:
it is necessary to:
1. start `@shutdown_consumer` at first
2. report `received` event.

It is because that travis-hub checks whether the build hasn't been already cancelled yet. If so, it sends back cancel event.